packages/swapper/src/swappers/ButterSwap/endpoints.test.ts

```diff
@@ -0,0 +1,187 @@
+import { describe, expect, it, vi } from 'vitest'
+
+import {
+  findToken,
+  getBuildTx,
+  getRoute,
+  getRouteAndSwap,
+  getSupportedChainList,
+  isBuildTxSuccess,
+  isRouteAndSwapSuccess,
+  isRouteSuccess,
+} from './xhr'
+
+vi.setConfig({ testTimeout: 10000 })
+
+describe('endpoints', () => {
+  describe('getSupportedChainList', () => {
+    it('should return a list of supported chains from the real API', async () => {
+      const result = await getSupportedChainList()
+      result.match({
+        ok: response => {
+          expect(response.errno).toBe(0)
+          expect(Array.isArray(response.data)).toBe(true)
+          expect(response.data.length).toBeGreaterThan(0)
+        },
+        err: error => {
+          expect.fail(JSON.stringify(error))
+        },
+      })
+    })
+  })
+
+  describe('findToken', () => {
+    it('should return token information from the real API', async () => {
+      const chainId = 1 // Ethereum
+      const address = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' // WETH
+      const result = await findToken(chainId, address)
+      result.match({
+        ok: response => {
+          expect(response.errno).toBe(0)
+          const token = response.data.find(
+            t => t.address.toLowerCase() === address.toLowerCase() && t.chainId === chainId,
+          )
+          expect(token).toBeDefined()
+        },
+        err: error => {
+          expect.fail(JSON.stringify(error))
+        },
+      })
+    })
+  })
+
+  describe('getRoute', () => {
+    it('should return a route from the real API', async () => {
+      const fromChainId = 137 // Polygon
+      const tokenInAddress = '0x2791bca1f2de4661ed88a30c99a7a9449aa84174' // USDC on Polygon
+      const toChainId = 137 // Polygon
+      const tokenOutAddress = '0x7ceb23fd6bc0add59e62ac25578270cff1b9f619' // WETH on Polygon
+      const amount = '1000000' // 1 USDC
+      const slippage = '150'
+      const result = await getRoute(
+        fromChainId,
+        tokenInAddress,
+        toChainId,
+        tokenOutAddress,
+        amount,
+        slippage,
+      )
+      result.match({
+        ok: response => {
+          expect(isRouteSuccess(response)).toBe(true)
+          if (!isRouteSuccess(response)) {
+            expect.fail(`Unexpected errno in ok branch: ${response.errno}`)
+            return
+          }
+          const route = response.data[0]
```

Comment:
q: are routes stable here? Could the order of routes (or the single found best route) change between quote and rate?
Noting we have `to`, `value` and `data` in `butterSwapTransactionMetadata`, wondering if this could become stale and send a Tx that's not executable anymore, doubting Butter has refunds systems in place.